gci/src/main/java/com/gcinterceptor/gci/SheddingThreshold.java

```diff
@@ -0,0 +1,53 @@
+package com.gcinterceptor.gci;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+public class SheddingThreshold {
+	private final long MIN_SHEDDING_THRESHOLD = 32 * 1024 * 1024; // TODO(David) Update this value, if needed
+	private final long MAX_SHEDDING_THRESHOLD = 512 * 1024 * 1024;  // TODO(David) Update this value, if needed 
+	private final double START_MAX_OVER_HEAD = (float) 0.1; // Maximum accepted Overhead (#shed/#processed)
+	private final long SMOOTH_FACTOR = 5;   // Smooth out the exponential decay.
+	// Looking at the function we can see where that at 22 it reaches the overhead of 0.001.
+	// https://www.wolframcloud.com/objects/danielfireman/gci_overhead_exp_decay
+	private final int MAX_GCS =  23; // TODO(David) Update this value, if needed
+	
+	private AtomicLong threshold;
+	private int numGCs;
+	
+	public SheddingThreshold() {
+		threshold = new AtomicLong((long) (MIN_SHEDDING_THRESHOLD + (Math.random() * MIN_SHEDDING_THRESHOLD)));
+	}
+	
+	public double getThreshold() {
```

Comment:
No need to Threshold word in this method. The class is already ShedingThreshold :) Same for the update method. Those should be get and update